### PR TITLE
[WIP] Adding jmp loss scaling support to Levanter

### LINKED ISSUE #122: Wire up jmp's loss scaling support for fp16
If we're gonna have to run on v100s, we're gonna need to support fp16 (😱). Jmp makes this reasonably easy to do: https://github.com/deepmind/jmp#loss-scaling

Needs to be wired up, put in configs as appropriate, and debugged.

### PR DESCRIPTION
**Pull request by alat-rights**
## Description
PR is intended to add jmp loss scaling to levanter to enable fp16. This should be a minor change.

This PR enables users of `next_token_loss` to pass in a `loss_scale` (jmp.LossScale) to scale the loss.

**TODO**: Apply `loss_scale.unscale` to grads.

This PR changes the following:
- Imports jmp in loss.py
- Adds the `loss_scale` parameter to `next_token_loss`
- Adds the `loss_scale` parameter to `cross_entropy_and_logsumexp_penalty`
- Scales `loss` with `loss_scale` in `cross_entropy_and_logsumexp_penalty`

Remaining steps:
- Apply `loss_scale.unscale` to grads.
- Write tests for new usage
- Update config
- Ensure tests pass. Debug as needed
- Update docs

## Fixes Issues
Resolves #122 

## Unit test coverage
No unit tests written yet.

## Known breaking changes/behaviors
Should not break anything.

### CODE REVIEW

**Comment by alat-rights** [comment]
Config: https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/config.py

**Comment by alat-rights** [comment]
Additional code points: 
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/callbacks.py#L30
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/main/lora_lm.py#L148
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/main/eval_lm.py#L69
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/main/train_lm.py#L126
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/models/lm_model.py#L76
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/grad_accum.py#L102
- https://github.com/stanford-crfm/levanter/blob/6ab19c26ffc2c92772475e5ac87126aece863d3f/src/levanter/main/viz_logprobs.py#L74
-

**Comment by dlwh** [comment]
hi @alat-rights thanks for taking this on. LMK if you would like to chat. If possible I'd like to keep the scale out of library loss functions, but i'm not sure what's the most ergonomics

**Comment by alat-rights** [comment]
> If possible I'd like to keep the scale out of library loss functions

I don't know the codebase well, but I do feel confused here. It seems to me that the desired action is "when anyone uses a loss function, they can also scaling, in which case their loss will be scaled appropriately."

I'd love to chat. I can reach out with contact information by email.
